README.md

```diff
@@ -1,2 +1,258 @@
+
+For more information about Vault One, you can check it out here: https://vaultone.com/
+
+These interactions are read-only, thus this library will not implement write operations using the VaultOne API. The necessary API endpoints are the following:
+  - Obtaining an access token to the API;
+  - Retrieving credentials based on a filter that identifies the credential;
+  - Acquiring a credential secret using a credential unique id provided by the VaultOne API.
+
+The API endpoints that are implemented in this first version are the following, and will be referred hereinafter by their corresponding names:
+  - POST /api/TokenAuth/Authenticate - Authenticate Request
+  - GET /api/services/app/Credentials/GetAllCredentials?Filter= - Get Credentials Request
+  - GET /api/services/app/Credentials/GetCredentialSecret?CredentialId= - Get Credential Secret Request
+
+This API Client implementation only returns the relevant information to be used in the other requests, such as the accessToken and the userId.
+
+## Get Credentials Request
+
+This API Client implementation returns the first credential that matches the filter and only returns relevant information to be used in the Get Credential Secret request, such as the credential id. In order to return the first credential, the function `get_credential_by_filter` should be called. 
+
+If the API user wishes to return all the credentials that match a filter, the function `get_all_credentials_by_filter` should be called.
+
+## Get Credential Secret Request
+
+The Get Credential Secret Request must send the "authorization" header with the token obtained from the Authenticate Request. The CredentialId query parameter should be the credential id obtained from the Get Credentials Request.
+
+This API Client implementation returns the username and the password associated with this credential id.
+
+The environment variables VAULT_ONE_API_USERNAME and VAULT_ONE_API_PASSWORD are required and they must be encoded in base64 format to satisfy Git CI/CD variables requirements.
+
+The environment VAULT_ONE_API_BASE_URL is also required, and its the VaultOne URL. It is used for performing the requests in the desired VaultOne client.
+
+## Get Credential Secret by Credential Name
+
+The function `get_credential_secret_by_credential_name` encapsulates the three requests, and returns the credential parameters (username and password) given a credential name. You may call the function `get_all_credential_secrets_by_credential_name` to obtain an array of credentials for this credential name instead of obtaining a single credential.
+
+## Example returns
+
+All of the functions that interact with the API endpoints previously described in this document return an object with 2 (two) fields: `data` and `status`. The `data` field may be an array, an object or `null`, depending on the function called and the response received by the API.
+
+### `authorize`
```

Comment:
we don't need this now.. but we could have created one python class to abstract these dict responses... @vavieira10